Controllers/EmployeeManagement/CouriersController.cs

```diff
@@ -63,15 +58,39 @@ public ActionResult<Courier> Get(int id)
         /// <param name="value">new created courier</param>
         /// <returns>one courier by id</returns>
         /// <response code="201">Returns the newly created item</response>
-        /// <response code="404">If the item is null</response>
+        /// <response code="400">If the item is null</response>
         // POST api/<CouriersController>
         [HttpPost]
         public ActionResult<Courier> Post([FromBody] Courier value)
         {
-            if (value != null)
-                return CreatedAtAction("Get", new { id = value.id }, value);
+            _context.Couriers.AddAsync(value);
 
-            return new StatusCodeResult(StatusCodes.Status404NotFound);
+            var checkResponse = Courier.CheckIfValid(value);
+
+            if (checkResponse == 1)
+                return BadRequest("Invalid body: values of Courier cannot be null");
+            else if (checkResponse == 2)
+                return BadRequest("Invalid body: invalid Courier Id");
+            else if (checkResponse == 3)
+                return BadRequest("Invalid body: invalid Courier Name");
+            else if (checkResponse == 4)
+                return BadRequest("Invalid body: invalid Courier Surname");
+            else if (checkResponse == 5)
+                return BadRequest("Invalid body: invalid Courier Email");
+            else if (checkResponse == 6)
+                return BadRequest("Invalid body: invalid Courier Username");
+            else if (checkResponse == 7)
+                return BadRequest("Invalid body: invalid Courier Password");
+            else if (checkResponse == 8)
+                return BadRequest("Invalid body: invalid Courier CreatedBy");
+            else if (checkResponse == 9)
+                return BadRequest("Invalid body: invalid Courier PhoneNurmber");
+            else if (checkResponse == 10)
+                return BadRequest("Invalid body: invalid Courier Transportation");
```

Comment:
This seems repetitive, maybe CheckIfValid could return a string with the message and null otherwise and you could just prepend "Invalid body: " to it?